### PR TITLE
Fix: Compilation with DEBUG_DUMP_COMMANDS enabled

### PR DESCRIPTION
**Pull request by LordAro**
## Motivation / Problem

#11601

## Description

Fix compile as appropriate -

* Fix from_chars to include both characters of the hex byte
* Fix next_date usage
* Stop trying to use CommandPacket::tile (just leave it as unused?)

## Limitations

I don't actually know what I'm doing here, but it does compile now. Probably needs @2TallTyler / @JGRennison and/or someone with actual commands.log experience to weigh in...

In particular, I feel like tile could probably be removed from the output entirely - feels like a bigger job though, idk

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, game_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: src/network/network.cpp [hunk 1]
```diff
@@ -1109,8 +1109,11 @@ void NetworkGameLoop()
 				int company;
 				uint cmd;
 				char buffer[256];
-				int ret = sscanf(p, "%x; %x; %x; %x; %x; %x; %255s", &next_date, &next_date_fract, &company, &cmd, &cp->err_msg, &cp->tile, buffer);
+				uint32_t next_date_raw;
```
Comment (2TallTyler) [inline src/network/network.cpp:1112]: TimerGameCalendar::Date is an int32_t. Same in the two `else if` blocks below.

**Comment by JGRennison** [comment]
When building locally it still complains about about the TileIndex, however this does not prevent the sscanf from working correctly.

```
/home/jgr/openttd/trunk/src/network/network.cpp: In function ‘void NetworkGameLoop()’:
/home/jgr/openttd/trunk/src/network/network.cpp:1114:75: warning: format ‘%x’ expects argument of type ‘unsigned int*’, but argument 8 has type ‘TileIndex*’ {aka ‘StrongType::Typedef<unsigned int, TileIndexTag, StrongType::Compare, StrongType::Integer, StrongType::Compatible<int>, StrongType::Compatible<long int> >*’} [-Wformat=]
 1114 |                                 int ret = sscanf(p, "%x; %x; %x; %x; %x; %x; %255s", &next_date_raw, &next_date_fract, &company, &cmd, &cp->err_msg, &tile, buffer);
      |                                                                          ~^                                                                          ~~~~~
      |                                                                           |                                                                          |
      |                                                                           unsigned int*                                                              TileIndex* {aka StrongType::Typedef<unsigned int, TileIndexTag, StrongType::Compare, StrongType::Integer, StrongType::Compatible<int>, StrongType::Compatible<long int> >*}

```

The dummy TileIndex can be omitted entirely by changing the sscanf string to use %*x instead of %x, as in:

```
int ret = sscanf(p, "%x; %x; %x; %x; %x; %*x; %255s", &next_date_raw, &next_date_fract, &company, &cmd, &cp->err_msg, buffer);
assert(ret == 6);
```

If the the dummy TileIndex is left in, the assert needs to be changed to `assert(ret == 7);`

Otherwise, it seems to work correctly when loading a commands.log file.

**Comment by LordAro** [comment]
Strange, didn't get a warning here. Ah well

Thoughts on just removing tile from the output entirely? If it's not being read in, is there any need for it to be there? I suppose it could help with manual debugging as it more clearly separates it out from the rest of the command parameters?

**Comment by rubidium42** [comment]
> Thoughts on just removing tile from the output entirely? If it's not being read in, is there any need for it to be there? I suppose it could help with manual debugging as it more clearly separates it out from the rest of the command parameters?

Seems fine to me. It's in `LogCommandExecution` which is called in the same function that calls `::NetworkSendCommand`. `::NetworkSendCommand` does not get `tile` passed, so I'm fairly certain that `tile` is not needed in the desync logs.

File: src/network/network.cpp (hunk 1 above)
Comment (rubidium42) [reply src/network/network.cpp:1112]: There's no hexadecimal format for signed integers, so either the format needs to be changed to '%d' and the output of the debug commands needs to be changed to not be written in hexadecimal, or we accept the debug commands are written hexadecimal and read them as unsigned numbers.

**Comment by LordAro** [comment]
I've gone with explicit int32_t casts for Date, and removed tile entirely from the desync debug output

**Review by rubidium42 — APPROVED** [review]

**Comment by LordAro** [comment]
@JGRennison can you confirm this all works now?

**Comment by JGRennison** [comment]
> @JGRennison can you confirm this all works now?

It seems to work fine here

**Pull request merged**